internal/uvm/cpugroups.go

```diff
@@ -2,55 +2,31 @@ package uvm
 
 import (
 	"context"
-	"errors"
-	"fmt"
 
 	"github.com/Microsoft/hcsshim/internal/cpugroup"
 	"github.com/Microsoft/hcsshim/internal/hcs/resourcepaths"
 	hcsschema "github.com/Microsoft/hcsshim/internal/hcs/schema2"
 )
 
 // ReleaseCPUGroup unsets the cpugroup from the VM
+// It does this by setting the VM's cpugroup to the null group ID of 00000000-0000-0000-0000-000000000000
+//
+// Since a VM must be moved to the null group before potentially being added to a different
+// cpugroup, that means there may be a segment of time that the VM's cpu usage runs unrestricted.
 func (uvm *UtilityVM) ReleaseCPUGroup(ctx context.Context) error {
-	groupID := uvm.cpuGroupID
-	if groupID == "" {
+	if uvm.cpuGroupID == "" {
 		// not set, don't try to do anything
 		return nil
 	}
-	if err := uvm.unsetCPUGroup(ctx); err != nil {
-		return fmt.Errorf("failed to remove VM %s from cpugroup %s", uvm.ID(), groupID)
-	}
-	return nil
-}
-
-// SetCPUGroup setups up the cpugroup for the VM with the requested id
-func (uvm *UtilityVM) SetCPUGroup(ctx context.Context, id string) error {
```

Comment:
So we can't set it at runtime anymore?

Reply:
From what I can tell this was just a workaround as we didn't support assigning it at creation time. I'd asked @katiewasnothere if we even needed to support runtime add anymore and there was no concerns but I can certainly leave this in.

Reply:
Yeah this was just a workaround. The expectation is that the orchestrator or host owner will create and manage cpugroups, so if they want to move a VM to a different cpugroup during runtime, they would have to handle it themselves, we don't provide knobs for that. 

I don't think we need to keep this workaround, but make sure to still have a build block for assigning during create.

Reply:
I think we can expect (at least for the time being) for cpugroup creation/config/deletion to be handled out-of-band, but it seems like assigning a VM to a cpugroup is something we should own. It seems reasonable to me that we keep this exposed.

Reply:
By assigning do you mean just the initial assignment or additional reassignments during runtime?

Reply:
The initial assignment seems the most important part, but I don't think there is any reason we can't expose both?

Reply:
The point of us supporting the initial assignment is just so we can ensure that the VM starts immediately in the target cpugroup and that the VM's memory and cpu resources come from the same numa node(s) on start. 

I don't think that exposing reassignment through us adds much and since the orchestrator is already in charge of managing the rest of the aspects of a cpugroup, I think it might be weird to have them call into us specifically for reassignment (this would need to be done through an update pod request). 

Another thing to note: CO+ builds [will] support directly moving VMs between cpugroups (iow no need to move to the null group and then to the target new build). 

Thoughts?

Reply:
Thanks for the background Kathryn :). It sounds like we don't need these really.. and afaik no ones even using this at the moment either. The runtime add/removes aren't actually hooked up to anything right? The functionality is here but there's nothing that someone could call externally to end up invoking any of this besides the previous way the initial assignment worked (add to group in `uvm.Start` and teardown in `uvm.Close`). I'm not tied to leaving or removing them, I don't see any harm in leaving them but if the plan was for reassignments to happen outside of the shim then I don't see a reason to keep them.

Reply:
If we don't think there is any need to support changing what cpugroup a VM is assigned to, I'm fine with taking this out. But if this is needed, I think we will need this, as there is no other path for the orchestrator to take.

Reply:
I removed it last night as from what Kathryn was saying it sounds like reassignments should be handled by whoever created the thing also. It would be trivial to re-add this as it wasn't actually hooked up to anything in the first place (but it would also be trivial to just leave it here and fix the linter warnings :P).

Reply:
aaaaand added it back per discussion :)